third-party/realdds/doc/DDS ICD.md

```diff
@@ -0,0 +1,147 @@
+    ],
+    "sensor-name":"Stereo Module",
+    "type":"ir"
+
+#### stream-options format example
+
+    "id":"stream-options",
+    "n-options":3,        //Integer, number of device options
+    "options":            //Array of supported options, the size of n-options
+    [
+        {"description":"Enable / disable backlight compensation","name":"Backlight Compensation","range-default":0.0,"range-max":1.0,"range-min":0.0,"range-step":1.0,"value":0.0},
+        {"description":"UVC image brightness","name":"Brightness","range-default":0.0,"range-max":64.0,"range-min":-64.0,"range-step":1.0,"value":0.0},
+        {"description":"UVC image contrast","name":"Contrast","range-default":50.0,"range-max":100.0,"range-min":0.0,"range-step":1.0,"value":50.0}
+    ],
+    "stream-name":"Infrared 1"
+
+### Instant notifications
+
+Other notifications are sent as needed, as a reply to **set-option** or **query-option**
+
+#### query-option reply format example
+
+    "id":"query-option",
+    "counter": 123,       //Integer, identical to the set-option command counter
+    "value": 12.34,       //Float
+    "successful":false,
+    "failure-reason":""
+
+#### set-option reply format example
+
+    "id":"set-option",
+    "counter": 123,       //Integer, identical to the set-option command counter
+    "successful":false,
+    "failure-reason":""
+
+## Control
+
+Controls are commands from the user to the camera.
+Using controls the user can open or close streaming, or he can query current option values or set new values affecting the camera behaviour.
+
+When using librealsense API commands will automatically be converted to the appropriate commands that will be sent to the camera over the DDS network.
+
+### open-streams command format example
+
+    "id":"open-streams",
+    "stream-profiles":{"Color":[30,"RGB8",848,480],"Depth":[30,"Z16",848,480]} //A list of streams to open and the requested profile for each one
+
+### close-streams command format example
+
+    "id":"close-streams",
+    "stream-names":["Depth","Color"] //A list of streams to close
+
+### query-option command format example
+
+    "id":"query-option",
+    "counter":107,        //Integer, unique identifier for this command
+    "option-name":"Auto Gain Limit Toggle",
+    "owner-name":"Depth"
+
+### set-option command format example
+
+    "id":"set-option",
+    "counter":108,        //Integer, unique identifier for this command
+    "option-name":"Enable Auto Exposure",
+    "owner-name":"Depth",
+    "value":0.0
```

Comment:
For all these, the format of the message itself is clear, and it is fantastic, but how should the user send these via our dds wrapper? (short code would be nice here too :-) )

Reply:
Code examples are out of scope for this document, they can be added later in another document